ASPXParser/SolutionData.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace ASPXParser
+{
+    public class SolutionData
+    {
+        public IDictionary<string, object> DataDictionary { get; set; }
+
+        public IEnumerable<string> AllFiles { get; set; }
+
+        public IEnumerable<string> AllWebFormsFiles { get; set; }
+
+        public SolutionData()
+        {
+            DataDictionary = new Dictionary<string, object>();
+            AllFiles = new List<string>();
+            AllWebFormsFiles = new List<string>();
+        }
+
+        public void GetData(string directoryPath)
+        {
+            AllFiles = Directory.EnumerateFiles(directoryPath, "*.*", SearchOption.AllDirectories);
+            AllWebFormsFiles = AllFiles.Where(s => s.EndsWith(".aspx")
+                                                   || s.EndsWith(".aspx.cs")
+                                                   || s.EndsWith(".ascx")
+                                                   || s.EndsWith(".ascx.cs")
+                                                   || s.EndsWith(".master")
+                                                   || s.EndsWith(".master.cs"))
+                .OrderByDescending(filename => filename);
+
+            AddProjectCountToDataDictionary(DataDictionary, AllFiles);
+            AddFileCountsToDataDictionary(DataDictionary, AllWebFormsFiles);
```

Comment:
from a coding standpoint this is clean.. but from algo stand point we can achieve all of the 3 counts in one iteration rather 3 iterations... :) see if you could find a balance

Reply:
Refactored file enumeration so we collect all counts in a single pass. Let me know if this new solution requires adjustments.